pkg/config/grpc_server.go

```diff
@@ -0,0 +1,50 @@
+package config
+
+import (
+	"math"
+	"time"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	UpdateRequestAction = "update_request"
+	DeleteRequestAction = "delete_request"
+	CreateRequestAction = "create_request"
+)
+
+type GRPCServerConfig struct {
+	EnableGRPCServer      bool          `json:"enable_grpc_server"`
+	Hostname              string        `json:"hostname"`
+	TLSCertFile           string        `json:"grpc_tls_cert_file"`
+	TLSKeyFile            string        `json:"grpc_tls_key_file"`
+	EnableTLS             bool          `json:"enable_grpc_tls"`
+	BindAddress           string        `json:"bind_address"`
+	MaxConcurrentStreams  uint32        `json:"max_concurrent_steams"`
+	MaxReceiveMessageSize int           `json:"max_receive_message_size"`
+	MaxSendMessageSize    int           `json:"max_send_message_size"`
+	ConnectionTimeout     time.Duration `json:"connection_timeout"`
+	WriteBufferSize       int           `json:"write_buffer_size"`
+	ReadBufferSize        int           `json:"read_buffer_size"`
+	MaxConnectionAge      time.Duration `json:"max_connection_age"`
+}
+
+func NewGRPCServerConfig() *GRPCServerConfig {
+	return &GRPCServerConfig{}
+}
+
+func (s *GRPCServerConfig) AddFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&s.EnableGRPCServer, "enable-grpc-server", false, "Enable gRPC server")
+	fs.StringVar(&s.BindAddress, "grpc-server-bindaddress", "localhost:8090", "gPRC server bind adddress")
+	fs.StringVar(&s.Hostname, "grpc-server-hostname", "", "gPRC Server's public hostname")
```

Comment:
do we want to start this as a separated binary, why do we need grpc/http hostname separately? same to bind address

Reply:
we do not want to start this as a separated binary. so remove the hostname for gRPC. for the bind address, we need to start gPRC in different port.

Reply:
can we have different bind-port flags instead of bind-address?